codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -270,6 +273,10 @@ public ChannelFuture writeWindowUpdate(ChannelHandlerContext ctx, int streamId,
     @Override
     public ChannelFuture writeFrame(ChannelHandlerContext ctx, byte frameType, int streamId, Http2Flags flags,
             ByteBuf payload, ChannelPromise promise) {
+        Http2Stream stream = streamId > 0 ? connection().stream(streamId) : null;
+        if (stream != null) {
+            stream.frameSent();
```

Comment:
This method says explicitly there are not state checks. So it is possible this frame is still in idle, but we still call `frameSent()`.